src/common/base/NebulaKeyUtils.cpp

```diff
@@ -86,6 +86,59 @@ std::string NebulaKeyUtils::kvKey(PartitionID partId, const folly::StringPiece&
     return key;
 }
 
+// static
+void NebulaKeyUtils::indexRaw(const IndexValues &values, std::string& raw) {
+    std::vector<int32_t> colsLen;
+    for (auto& col : values) {
+        if (col.first == cpp2::SupportedType::STRING) {
+            colsLen.emplace_back(col.second.size());
+        }
+        raw.append(col.second.data(), col.second.size());
+    }
+    for (auto len : colsLen) {
```

Comment:
why append column size at the end of encoded-value?

Reply:
Used for variable-length column type ， for example `string`  .
if an index row likes string('abc') + string('abc') , when the where expression is col1 == 'abca', the current row should be skip.